shared/middleware/helpers/property_bag.rb

```diff
@@ -114,6 +116,12 @@ def set(name, value, ip_address, time = DateTime.now)
       },
     )
     value
+  rescue Aws::DynamoDB::Errors::ValidationException => e
+    if e.message == 'Item size has exceeded the maximum allowed size'
+      too_large
```

Comment:
My recommendation is to return a value indicating the type of error here, and to raise TooLarge from within PropertiesApi. There are a few helpers that have been used from elsewhere within ruby code and each time we have had to extract response logic out of the helper to make it possible, so I'd prefer not to introduce more knowledge of http responses here.